pkg/kudoctl/kudoinit/migration/01_migrate_ov_names/migrate_ov_names.go

```diff
@@ -0,0 +1,119 @@
+//nolint:golint,stylecheck
+package _01_migrate_ov_names
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kudoapi "github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/clog"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/kube"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/kudoinit/migration"
+)
+
+var (
+	_ migration.Migrater = &MigrateOvNames{}
+)
+
+type MigrateOvNames struct {
+	client *kube.Client
+	dryRun bool
+	ctx    context.Context
+
+	migratedOVs map[string]*kudoapi.OperatorVersion
+}
+
+func New(client *kube.Client, dryRun bool) *MigrateOvNames {
+	return &MigrateOvNames{
+		client:      client,
+		dryRun:      dryRun,
+		ctx:         context.TODO(),
+		migratedOVs: map[string]*kudoapi.OperatorVersion{},
+	}
+}
+
+func (m *MigrateOvNames) CanMigrate() error {
+	// No migrate check required for this migration
+	return nil
+}
+
+func (m *MigrateOvNames) Migrate() error {
+	if !m.dryRun {
+		clog.V(0).Printf("Migrate OperatorVersion names")
+	}
+	return migration.ForEachNamespace(m.client, m.migrateInNamespace)
+}
+
+func (m *MigrateOvNames) migrateInNamespace(ns string) error {
```

Comment:
so this will be run when you upgrade to 0.17 but will this be run again when migrating from 0.17 to 0.18? I don't think we want to do that

Reply:
It would, yes - And it doesn't matter, the migrations are (or should be) idempotent. You can re-run them again and again, they shouldn't modify anything after the first run.